src/data/list/chain.lean

```diff
@@ -228,44 +234,44 @@ theorem chain'.imp_head {x y} (h : ∀ {z}, R x z → R y z) {l} (hl : chain' R
 hl.tail.cons' $ λ z hz, h $ hl.rel_head' hz
 
 theorem chain'_reverse : ∀ {l}, chain' R (reverse l) ↔ chain' (flip R) l
-| [] := iff.rfl
-| [a] := by simp only [chain'_singleton, reverse_singleton]
+| []            := iff.rfl
+| [a]           := by simp only [chain'_singleton, reverse_singleton]
 | (a :: b :: l) := by rw [chain'_cons, reverse_cons, reverse_cons, append_assoc, cons_append,
     nil_append, chain'_split, ← reverse_cons, @chain'_reverse (b :: l), and_comm, chain'_pair, flip]
 
 theorem chain'_iff_nth_le {R} : ∀ {l : list α},
   chain' R l ↔ ∀ i (h : i < length l - 1),
     R (nth_le l i (lt_of_lt_pred h)) (nth_le l (i+1) (lt_pred_iff.mp h))
-| [] := by simp
-| (a :: nil) := by simp
+| []            := by simp
+| [a]           := by simp
 | (a :: b :: t) :=
 begin
   rw [chain'_cons, chain'_iff_nth_le],
   split,
-  { rintros ⟨R, h⟩ i w,
+  { rintro ⟨R, h⟩ i w,
     cases i,
     { exact R, },
     { convert h i _ using 1,
       simp only [succ_eq_add_one, add_succ_sub_one, add_zero, length, add_lt_add_iff_right] at w,
       simpa using w, },
    },
-  { rintros h, split,
+  { rintro h, split,
     { apply h 0, simp, },
     { intros i w, convert h (i+1) _ using 1,
       simp only [add_zero, length, add_succ_sub_one] at w,
       simpa using w, }
     },
 end
 
-/-- If `l₁ l₂` and `l₃` are lists and `l₁ ++ l₂` and `l₂ ++ l₃` both satisfy
-  `chain' R`, then so does `l₁ ++ l₂ ++ l₃` provided `l₂ ≠ []` -/
+/-- If `l₁ l₂` and `l₃` are lists and `l₁++l₂` and `l₂++l₃` both satisfy
+  `chain' R`, then so does `l₁++l₂++l₃` provided `l₂ ≠ []` -/
```

Comment:
Why are you adding spaces around `::` but removing the around `++`? That might be confusing, since `l₁++a :: l₂` is parsed as `l₁++(a :: l₂)`. I'd prefer spaces around both.

bors d+

Reply:
Whoops, I meant to add them around everything. Fixing it.